lib/acme/client/resources/authorization.rb

```diff
@@ -1,44 +1,86 @@
+# frozen_string_literal: true
+
 class Acme::Client::Resources::Authorization
-  HTTP01 = Acme::Client::Resources::Challenges::HTTP01
-  DNS01 = Acme::Client::Resources::Challenges::DNS01
-  TLSSNI01 = Acme::Client::Resources::Challenges::TLSSNI01
+  attr_reader :url, :expires, :status, :wildcard, :attributes
+
+  def self.arguments_from_response(response)
+    attributes = response.body
+    status = attributes.fetch('status')
+    expires =  Time.parse(attributes.fetch('expires'))
+    challenges = attributes.fetch('challenges')
+    wildcard = attributes.fetch('wildcard', false)
 
-  attr_reader :client, :uri, :domain, :status, :expires, :http01, :dns01, :tls_sni01
+    {
+      status: status,
+      expires: expires,
+      challenges: challenges,
+      wildcard: wildcard
+    }
+  end
 
-  def initialize(client, uri, response)
+  def initialize(client, **arguments)
     @client = client
-    @uri = uri
-    assign_attributes(response.body)
+    assign_attributes(arguments)
+  end
+
+  def deactivate
+    assign_attributes **@client.deactivate_authorization(url: url).to_h
+    true
+  end
+
+  def reload
+    assign_attributes **@client.authorization(url: url).to_h
+    true
+  end
+
+  def challenges
+    @challenges.map do |challenge|
+      initalize_challenge(challenge)
+    end
+  end
+
+  def http01
+    @http01 = challenges.find { |challenge|
+      challenge.is_a?(Acme::Client::Resources::Challenges::HTTP01)
+    }
```

Comment:
There is a mix of styles using `do; end` vs. `{ }`. Should this and `#dns01` use `do; end`?

Reply:
There is a bug here because I intended to do `||=`.

On this project I've been following the convention of using `{ }` when the return value and `do / end` when the side effect is desired.